Userland/Libraries/LibWeb/Painting/SVGPathPaintable.cpp

```diff
@@ -65,10 +65,9 @@ void SVGPathPaintable::paint(PaintContext& context, PaintPhase phase) const
     auto const* svg_node = layout_box().first_ancestor_of_type<Layout::SVGSVGBox>();
     auto svg_element_rect = svg_node->paintable_box()->absolute_rect();
 
-    // FIXME: This should not be trucated to an int.
```

Comment:
Doesn't this still lose subpixel information? Rounding is better than flooring, but passing through a float is probably better still, no?

Reply:
(Like, imagine the same test case, but with width/height dimensions < viewbox width/height, so that there's a non-integer scaling factor on the svg)

Reply:
I.e. use this instead:

```
    auto offset = svg_element_rect.location().to_type<float>() * static_cast<float>(context.device_pixels_per_css_pixel());
```

Does that make sense?

Reply:
I'll check if this still gives the right results, and if so will mention you as co-author!

Reply:
I've tried a couple of variations on this and none of them get me close to the result other browsers show. Using the exact offset puts the left side of the rectangle wholly outside of the bounding rects with Skia, even if I expand them. Do you have an idea as to where we might be doing something wrong?

Reply:
Hm! Does this help?

```diff
diff --git a/Userland/Libraries/LibWeb/Painting/PaintContext.cpp b/Userland/Libraries/LibWeb/Painting/PaintContext.cpp
index d9753c55411..32e0e1eb7af 100644
--- a/Userland/Libraries/LibWeb/Painting/PaintContext.cpp
+++ b/Userland/Libraries/LibWeb/Painting/PaintContext.cpp
@@ -62,12 +62,11 @@ DevicePixelPoint PaintContext::floored_device_point(CSSPixelPoint point) const
 
 DevicePixelRect PaintContext::enclosing_device_rect(CSSPixelRect rect) const
 {
-    return {
-        floor(rect.x().to_double() * m_device_pixels_per_css_pixel),
-        floor(rect.y().to_double() * m_device_pixels_per_css_pixel),
-        ceil(rect.width().to_double() * m_device_pixels_per_css_pixel),
-        ceil(rect.height().to_double() * m_device_pixels_per_css_pixel)
-    };
+    auto x1 = floor(rect.x().to_double() * m_device_pixels_per_css_pixel);
+    auto y1 = floor(rect.y().to_double() * m_device_pixels_per_css_pixel);
+    auto x2 = ceil(rect.x().to_double() + rect.width().to_double() * m_device_pixels_per_css_pixel);
+    auto y2 = ceil(rect.y().to_double() + rect.height().to_double() * m_device_pixels_per_css_pixel);
+    return {x1, y1, x2 - x1, y2 - y1 };
 }
```

With serenity's path renderer I need both that and the offset above for the output of your test to not be clipped or look weird.

(I don't have a ladybird build setup, due to the brew requirement, so I can't test myself, sorry.)

Reply:
Ladybird with both changes:
![image](https://github.com/user-attachments/assets/bcd941df-ab7d-443f-8074-432861c2c818)

Maybe we mess something up in translating it to Skia.